Header header 

float32[] x_cur
float32[] y_cur
float32[] z_cur

float32[] x_old
float32[] y_old
float32[] z_old